mail-filter/postforward: Initial ebuild

Philippe23 opened the pull request:
- Create initial ebuild
 - Work around upstream not putting their .go files in src/ which
   makes "go build" angry
 - Work around golang-build complaining about pkg not existing
   and just do the install by hand
 - Patch default sendmail path to respect EPREFIX and use the
   standard postfix install location instead of depending on
   it being in the path

Package-Manager: Portage-2.3.8, Repoman-2.3.3

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Please convert it to EAPI 6.

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+inherit eutils golang-build
+
+DESCRIPTION="Postfix Sender Rewriting Scheme forwarding agent"
+#EGO_PN="github.com/zoni/${PN}"
```
Remove this line please.

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
This line can go.

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+	mail-filter/postsrsd"
+
+src_prepare() {
+	epatch "${FILESDIR}/${PN}_apply-sendmail-path.patch"
```
Define a PATCH array outside this function in which you'll add this file, like so:
```
PATCH=( "${FILESDIR}/${PN}_apply-sendmail-path.patch" )
```

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+src_prepare() {
+	epatch "${FILESDIR}/${PN}_apply-sendmail-path.patch"
+	sed -i -e "s/@GENTOO_PORTAGE_EPREFIX@/${EPREFIX}/" *.go
```
missing `|| die`

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+	# go build assumes files will be in src dir, but
+	# source files are in root in this package, so move
+	# them.
+	NEW_SRC_DIR="${S}/src/${EGO_PN}"
```
`export` ?

Philippe23 replied:
Is `local` kosher?

monsieurp commented on mail-filter/postforward/postforward-1.0.1.ebuild:
```diff
@@ -0,0 +1,45 @@
+# Standard golang-build src_install complains about pkg not
+# existing, so we go custom.
+src_install() {
+	dodoc README.md
```
call `einstalldocs` instead and remove this line.

Philippe23 replied:
Fine to just call `default` instead?

Philippe23 replied:
Nix that question, `default` calls `make install` which doesn't work for this package.  Forgot the whole reason I added the custom `src_install`....

Philippe23 commented:
Just wanted to say I'll probably get back to this later in the month.  Holidays, work, and a trip are getting in the way of me having a chance to work on this.

monsieurp commented:
Sure.

mgorny commented:
Ping².

Philippe23 commented:
Updated to implement @monsieurp's requests.  Assumed he wanted the changes squashed into the original from the "improve the commit message" statement.  Also updated version to 1.1.0 since that has been released in the interim.

Build log: [postforward-1.1.0_20180313-153638.log](https://github.com/gentoo/gentoo/files/1807610/postforward-1.1.0_20180313-153638.log)

Gentoo bug: https://bugs.gentoo.org/650398

mgorny commented on mail-filter/postforward/Manifest:
```diff
@@ -0,0 +1 @@
+DIST postforward-1.1.0.tar.gz 5707 SHA256 a2968dbc6374715d503c4efba67ee587d33a045a747d74253c4dd9f43df6fffa SHA512 39d7a33c454b1529a691b24fd69b311ba368caf59168d39146654cfe319ebc3cf73835e66e6c95a43747f389a6b954ac8da0d7e934e562c16cb85dc20c0f4804 WHIRLPOOL 329c42f424497f3b6dee433eab775752c3e59d37a8139f1e9f39eef9e749971a94321b98c88c6bd76144003c39116a26cf1386b7796e913494bffa689f3f6dd5
```
Those are wrong hashes. Please remove the Manifest file and make `repoman manifest` regenerate them.

mgorny commented on mail-filter/postforward/postforward-1.1.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+	# go build assumes files will be in src dir, but
+	# source files are in root in this package, so move
+	# them.
+	local NEW_SRC_DIR="${S}/src/${EGO_PN}"
```
The usual convention is to name global variables in uppercase but local variables in lowercase.

mgorny commented on mail-filter/postforward/postforward-1.1.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+	# Freak out if there's already something there because
+	# it means the package has changed and we'll need to
+	# adjust to it.
+	test ! -e "${NEW_SRC_DIR}" || die "${NEW_SRC_DIR} already exists"
```
Don't use `test` in bash, use `[[ ... ]]`. Also note that `[[ ... ]]` normally doesn't require quoting.

Philippe23 commented:
Rebuilt manifest, changed case of local variable, and switched away from `test` per @mgorny's request.  Left quotes on for consistency because they're needed in the other two uses and don't hurt in the `[[ ]]` use case.
